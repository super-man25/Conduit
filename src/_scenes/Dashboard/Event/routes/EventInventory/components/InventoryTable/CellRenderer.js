// @flow
import * as React from 'react';
import { isDefined } from '_helpers';
import { Flex } from '_components';

type Props = {
  cellData: any
};

export const defaultCellRenderer = (props: Props) => {
  const { cellData } = props;
  const value = isDefined(cellData) ? String(cellData) : '';

  return (
    <Flex align="center" marginLeft="1.25rem">
      {value}
    </Flex>
  );
};
