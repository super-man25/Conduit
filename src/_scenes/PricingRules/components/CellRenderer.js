// @flow
import * as React from 'react';
import { Flex, NumberInputField } from '_components';
import { EditPricingRuleInput } from './styled';

type Props = {
  cellData: any,
  isEditing: boolean,
  rulePropertyValue: string,
  columnData: any,
  updatePriceRuleProperty: (any) => void,
};

type State = {
  rulePropertyValue: string,
};

export class DefaultCellPresenter extends React.Component<Props, State> {
  state = { rulePropertyValue: '' };

  onRulePropertyChange = (e: SyntheticEvent<HTMLInputElement>) => {
    const { value } = (e.currentTarget: HTMLInputElement);
    const { updatePriceRuleProperty } = this.props;

    // convert empty string values to null
    const newPriceRuleProperty = value === '' ? null : value;

    updatePriceRuleProperty(newPriceRuleProperty);
  };

  render() {
    const {
      isEditing,
      rulePropertyValue,
      columnData: { displayFn },
    } = this.props;

    const value = rulePropertyValue || '';

    if (isEditing) {
      return (
        <Flex align="center">
          <NumberInputField
            component={EditPricingRuleInput}
            value={value}
            onChange={this.onRulePropertyChange}
            placeholder={'-'}
          />
        </Flex>
      );
    }
    return <Flex align="center"> {displayFn(value)} </Flex>;
  }
}
