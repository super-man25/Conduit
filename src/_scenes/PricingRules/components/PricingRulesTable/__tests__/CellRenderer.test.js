import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { DefaultCellPresenter } from '../CellRenderer';

describe('defaultCellRenderer', () => {
  const props = {
    updatePriceRuleProperty: jest.fn(),
    cellData: 20,
    isEditing: false,
    columnData: {
      displayFn: jest.fn((a) => a),
    },
  };

  it('should render correctly', () => {
    const wrapper = shallow(<DefaultCellPresenter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when editing', () => {
    const wrapper = shallow(
      <DefaultCellPresenter {...props} isEditing={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger onRulePropertyChange', () => {
    const fn = jest.fn();

    const wrapper = mount(
      <DefaultCellPresenter
        {...props}
        isEditing={true}
        updatePriceRuleProperty={fn}
      />
    );

    wrapper.find('input').simulate('change');

    expect(fn).toBeCalled();

    wrapper.unmount();
  });
});
