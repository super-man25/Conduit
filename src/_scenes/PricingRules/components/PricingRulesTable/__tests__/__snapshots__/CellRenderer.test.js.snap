// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultCellRenderer should render correctly 1`] = `
<Flex
  align="center"
>
   
   
</Flex>
`;

exports[`defaultCellRenderer should render correctly when editing 1`] = `
<Flex
  align="center"
>
  <NumberInputField
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          [Function],
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-kUaPvJ",
          "isStatic": false,
          "rules": Array [
            "
  position: relative;
  padding: 15px;
  font-size: 14px;
  border: 2px solid ",
            "#0A169A",
            ";
  border-color: ",
            [Function],
            ";
  border-radius: 3px;
  background: ",
            [Function],
            ";
  ::placeholder {
    color: ",
            "#999999",
            ";
  }
",
            "
  display: block;
  background: ",
            [Function],
            ";
  background-image: ",
            [Function],
            ";
  background-repeat: no-repeat;
  background-position: right center;
  &:-webkit-autofill {
    animation-name: ",
            [Function],
            ";
    animation-fill-mode: both;
  }
  width: ",
            [Function],
            ";
",
            "
  margin: 0 0.5rem 0 0;
  padding: 10px 8px;
  max-height: 35px;
  max-width: 80px;
",
          ],
        },
        "displayName": "Styled(Styled(styled.input))",
        "foldedComponentIds": Array [
          "sc-kvZOFW",
          "sc-jbKcbu",
        ],
        "render": [Function],
        "styledComponentId": "sc-kUaPvJ",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    onChange={[Function]}
    placeholder="-"
    value=""
  />
</Flex>
`;
